module/VuFindSearch/src/VuFindSearch/Command/GetUniqueKeyCommand.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * Get the backend's unique search key field (currently only supported by Solr)
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+namespace VuFindSearch\Command;
+
+use VuFindSearch\Backend\BackendInterface;
+
+/**
+ * Get the backend's unique search key field (currently only supported by Solr)
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class GetUniqueKeyCommand extends AbstractBase
+{
+    /**
+     * Execute command on backend.
+     *
+     * @param BackendInterface $backend Backend
+     *
+     * @return CommandInterface Command instance for method chaining
+     */
+    public function execute(BackendInterface $backend): CommandInterface
+    {
+        if (is_callable([$backend, 'getConnector'])) {
+            $connector = $backend->getConnector();
+            if (is_callable([$connector, 'getUniqueKey'])) {
+                return $this->finalizeExecution($connector->getUniqueKey());
```

Comment:
Should there be a `GetConnectorCommand`? Perhaps not, and then all connector method calls should be commands like this one here? Or if there is a `GetConnectorCommand`, would it be ok to just make direct connector method calls after obtaining the connector? Or should there be connector method specific commands as well?

In any case I guess we want to avoid a situation where there is a `GetConnectorCommand` and also `$backend->getConnector()` calls in other commands...

Reply:
My thinking on this is that the connector is an implementation detail of the backend; there is no requirement that a backend have a connector, it is simply a pattern that is common and useful. There is also nothing like a ConnectorInterface, so there's no guarantee of connector consistency across backends. I think it makes more sense for commands to focus on retrieving specific pieces of information rather than on gaining access to specific pieces of functionality, and if we ever need to expand something like this to apply to more backends, we just need to add an interface and move more logic out of the command and into the relevant backend(s). (I think this same philosophy applies to your work in #1886, where you wrote a command to returns the results of analysis rather than retrieving the object that does the analysis... which is also a reminder that we should get that finished up soon :-) ). Since the backend code already makes heavy use of the connector all over the place, trying to create an event hook to intercept all those calls would be a losing battle anyway. I think if you want to override connector behavior, you would be much better off doing it with a custom factory rather than a custom event listener. Would you agree, or am I missing something?

Reply:
Yes, I agree. And thanks for the reminder! 😁

Reply:
Great, thanks! I believe this is ready to merge now! Onward to make more progress on #2036...